ts/utils/__tests__/faq.test.ts

```diff
@@ -0,0 +1,58 @@
+import I18n from "../../i18n";
+import { FAQsCategoriesType, FAQType, getFAQsFromCategories } from "../faq";
+
+describe("getFAQsFromCategories", () => {
+  const categories: ReadonlyArray<FAQsCategoriesType> = [
+    "landing_SPID",
+    "landing_CIE",
+    "authentication_IPD_selection"
+  ];
+  const faqs: ReadonlyArray<FAQType> = [
+    {
+      title: I18n.t("faq.1.title"),
+      content: I18n.t("faq.1.content")
+    },
+    {
+      title: I18n.t("faq.2.title"),
+      content: I18n.t("faq.2.content")
+    },
+    {
+      title: I18n.t("faq.3.title"),
+      content: I18n.t("faq.3.content")
+    },
+    {
+      title: I18n.t("faq.4.title"),
+      content: I18n.t("faq.4.content")
+    },
+    {
+      title: I18n.t("faq.5.title"),
+      content: I18n.t("faq.5.content")
+    },
+    {
+      title: I18n.t("faq.6.title"),
+      content: I18n.t("faq.6.content")
+    },
+    {
+      title: I18n.t("faq.7.title"),
+      content: I18n.t("faq.7.content")
+    },
+    {
+      title: I18n.t("faq.8.title"),
+      content: I18n.t("faq.8.content")
+    }
+  ];
+
+  it("should return the FAQType array from an array of FAQsCategoriesType", () => {
+    expect(getFAQsFromCategories(categories)).toStrictEqual(faqs);
+  });
+
+  it("should return the FAQType array from an array of FAQsCategoriesType", () => {
+    expect(getFAQsFromCategories([...categories, ...categories])).toStrictEqual(
+      faqs
+    );
+  });
+
+  it("Should return an empty array", () => {
+    expect(getFAQsFromCategories([])).toStrictEqual([]);
+  });
```

Comment:
If I run this test it fails cause to an exception

```typescript
it("Should return an empty array", () => {
    expect(
      getFAQsFromCategories(["not mapped key" as FAQsCategoriesType])
    ).toStrictEqual([]);
  });
```
this happens when:
- we add a string literal in the **FAQsCategoriesType** union type
- we forgot to add that key in the **FAQs** mapping

So this says that implementation has a weakness.
Could it be fixed as follows?

```typescript
export const getFAQsFromCategories = (
  categories: ReadonlyArray<FAQsCategoriesType>
): ReadonlyArray<FAQType> => {
  const faqIDs: Set<number> = categories.reduce((acc, val) => {
    const ids = FAQs[val]; // could be undefined
    return new Set<number>([...acc, ...fromNullable(ids).fold([], s => s)]);
  }, new Set<number>());

  return Array.from(faqIDs).map<FAQType>(id => {
    return {
      title: I18n.t(`faq.${id}.title`),
      content: I18n.t(`faq.${id}.content`)
    };
  });
};

```

Furthermore we can fix also the FAQs type
```typescript
export const FAQs: {
  [key in FAQsCategoriesType]: ReadonlyArray<number> | undefined
}
```

Reply:
I just thought that it could be an error to be identified at compile time, i think it's good to add this test too!

I just refactored the fix proposal con **getFAQsFromCategories** as it follows:
```javascript
export const getFAQsFromCategories = (
  categories: ReadonlyArray<FAQsCategoriesType>
): ReadonlyArray<FAQType> => {
  const faqIDs: Set<number> = categories.reduce(
    (acc, val) =>
      new Set<number>([...acc, ...fromNullable(FAQs[val]).fold([], s => s)]),
    new Set<number>()
  );

  return Array.from(faqIDs).map<FAQType>(id => {
    return {
      title: I18n.t(`faq.${id}.title`),
      content: I18n.t(`faq.${id}.content`)
    };
  });
};
```

What do you think?